riscv/triggers.h

```diff
@@ -91,6 +91,8 @@ class trigger_t {
 protected:
   static action_t legalize_action(reg_t val, reg_t action_mask, reg_t dmode_mask) noexcept;
   bool common_match(processor_t * const proc) const noexcept;
+  // Same as common_match, but based on previous privilege
+  bool trap_match(processor_t * const proc) const noexcept;
```

Comment:
We only call the common_match() twice and the trap_match() only once. We may remove (inline) the common_match() and trap_match() functions because those function names barely present additional messages.